sparkles/tests/test_review.py

```diff
@@ -182,3 +184,113 @@ def test_run_aca_review_function():
+    acar.get_roll_options()
+    assert Quat(kw['att']).roll <= acar.roll_info['roll_max']
+    assert Quat(kw['att']).roll >= acar.roll_info['roll_min']
+
+
+def test_calc_targ_from_aca():
+    """
+    Confirm _calc_targ_from_aca seems to do the right thing based on obsid
+    This does a bit too much processing for what should be a lightweight test.
+    """
+    obs_kwargs = KWARGS_48464
+    aca_er = get_aca_catalog(**obs_kwargs)
+    acar_er = aca_er.get_review_table()
+    q_targ = acar_er._calc_targ_from_aca(acar_er.att, 0, 0)
+    # Cheat and use the string reprs for comparison to 8 decimals
+    assert Quat(obs_kwargs['att']).__repr__() == q_targ.__repr__()
+
+    or_obs_kwargs = obs_kwargs.copy()
+    or_obs_kwargs['obsid'] = 1
+    aca_or = get_aca_catalog(**or_obs_kwargs)
+    acar_or = aca_or.get_review_table()
+    q_targ = acar_or._calc_targ_from_aca(acar_or.att, 0, 0)
+    # Assert that the the OR version *doesn't* match and is off by about 70 arcsecs yaw
+    assert Quat(obs_kwargs['att']).__repr__() != q_targ.__repr__()
+    assert abs(Quat(obs_kwargs['att']).dq(q_targ).yaw * 3600) - 69.59 < .01
+
+
+def test_get_roll_intervals():
+    """
+    Test that the behavior of get_roll_intervals is different for ORs and ERs with
+    regard to use of offsets.  They are set to arbitrary large values in the test.
+    """
+
+    # This uses the catalog at KWARGS_48464, but would really be better as a fully
+    # synthetic test
+    obs_kwargs = KWARGS_48464
+    aca_er = get_aca_catalog(**obs_kwargs)
+    acar_er = aca_er.get_review_table()
+
+    kw_or = obs_kwargs.copy()
+    # Set this one to have an OR obsid (and not 0 which is special)
+    kw_or['obsid'] = 1
+    aca_or = get_aca_catalog(**kw_or)
+    acar_or = aca_or.get_review_table()
+
+    # Use these values to override the get_roll_intervals ranges to get more interesting
+    # outputs.  y_off and z_off are really 0 everywhere for now from ORViewer though.
+    y_off = 20 / 60.
+    z_off = 30 / 60.
+    roll_dev = 5
+
+    er_roll_intervs, er_info = acar_er.get_roll_intervals(
+        acar_er.get_candidate_better_stars(),
+        roll_dev=roll_dev, y_off=y_off, z_off=z_off)
+
+    or_roll_intervs, or_info = acar_or.get_roll_intervals(
+        acar_or.get_candidate_better_stars(),
+        roll_dev=roll_dev, y_off=y_off, z_off=z_off)
+
+    assert Quat(obs_kwargs['att']).roll <= er_info['roll_max']
+    assert Quat(obs_kwargs['att']).roll >= er_info['roll_min']
+
+    # The roll ranges in ACA rolls should be the same for both the ER and the OR version
+    assert or_info == er_info
+
+    # Up to this point this is really a weak functional test.  The following asserts
+    # are more regression tests for the attitude at obsid 48464
+    or_rolls = [interv['roll'] for interv in or_roll_intervs]
+    er_rolls = [interv['roll'] for interv in er_roll_intervs]
+    assert or_rolls != er_rolls
+
+    assert or_roll_intervs == [{'roll': 281.63739755173594,
```

Comment:
These exact comparisons are asking for trouble.  Define `exp_intervs` as what you've written and then:
```
for interv, exp_interv in zip(or_roll_intervs, exp_intervs):
    assert interv.keys() == exp_interv.keys()
    for key in interv.keys():
        if key.startswith('roll'):
            assert np.isclose(..., atol=1e-6, rtol=0)
        else:
            assert .. a == b
```

Reply:
I made a quick method to do this to not repeat the loop for the OR and ER cases.